app/lib/surt/canonicalize.rb

```diff
@@ -34,12 +40,25 @@ module Surt::Canonicalize
     /^(.*\/)(\([0-9a-z]{24}\)\/)([^\?]+\.aspx.*)$/i
   ].freeze
 
+  # TODO: should we refactor this into a more readable format? e.g:
```

Comment:
Meh. It's not to difficult to divine the intent here.

Reply:
I do worry about ones that check multiple params in combination, e.g:

```
/^(.*)(?:cfid=[^&]+&cftoken=[^&]+)(?:&(.*))?$/i
```

…that are susceptible to missing their targets if they’re out of order. But if so, we’re no worse than archive.org is. In either case, it’s no big deal right now.